packages/salesforcedx-vscode-core/src/commands/forceVisualforceComponentCreate.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { notificationService } from '../notifications';
+import { CancellableStatusBar, taskViewService } from '../statuses';
+import glob = require('glob');
+import { nls } from '../messages';
+import {
+  CancelResponse,
+  CompositeParametersGatherer,
+  ContinueResponse,
+  DirFileNameSelection,
+  ParametersGatherer,
+  SelectDirPath,
+  SelectFileName,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+const VF_CMP_EXTENSION = '.component';
+
+class ForceVisualForceComponentCreateExecutor extends SfdxCommandletExecutor<
+  DirFileNameSelection
+> {
+  public build(data: DirFileNameSelection): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_visualforce_component_create_text'))
+      .withArg('force:visualforce:component:create')
+      .withFlag('--componentname', data.fileName)
+      .withFlag('--label', data.fileName)
+      .withFlag('--outputdir', data.outputdir)
+      .build();
+  }
+
+  public execute(response: ContinueResponse<DirFileNameSelection>): void {
```

Comment:
Lines 57-78 will most likely be consistent behavior for the creation commands and replaces the default call of notificationService.reportCommandExecutionStatus. I was thinking either: 
1) Have a subclass of SfdxCommandletExecutor (SfdxCreateCommandletExecutor) to remove overriding execute in every create command class
2) Create a method in SfdxCommandletExecutor (reportExecution) that defaults to calling notificationService.reportCommandExecutionStatus and can be overridden. Then extract the subscribe + open file/reportExecutionError into a common helper method that all the classes will use to override reportExecution